### PR TITLE
fix: batch create collab api endpoint might cause tokio runtime issue

### PR DESCRIPTION
**Pull request by appflowy**
The batch/collab API endpoint is taking too long to complete, primarily because the decompression step is causing performance issues with the Tokio runtime. To address this:

1. remove the old version of the API endpoint.
2. optimize the decompression process by offloading it to Rayon, which will handle the decompression in parallel, improving performance and preventing the Tokio runtime from being overwhelmed by CPU-bound tasks.

### CODE REVIEW

File: src/api/workspace.rs [hunk 1]
```diff
@@ -583,68 +578,72 @@ async fn batch_create_collab_handler(
...
     }
   }
-  let duration = start_time.elapsed();
-  event!(
-    tracing::Level::DEBUG,
-    "end decompressing collab params list, time taken: {:?}",
-    duration
-  );
+  // Perform decompression and processing in a Rayon thread pool
+  let mut collab_params_list = tokio::task::spawn_blocking(move || {
```
Comment (Horusiath) [inline src/api/workspace.rs:613]: You're still spawning a thread in here.

**Review by Horusiath — COMMENTED** [review]
This is something that probably should be measured, at least in performance tests. Do we know how much size does Brotli saves us (after all it's optimised for HTML data not regular Yrs format), how long does it actually take and how other algorithms (ie. zstd) perform in comparison?

File: src/api/workspace.rs (hunk 1 above)
Comment (appflowy) [reply src/api/workspace.rs:613]: without using the spawn_blocking, it might block the tokio worker thread, causing the runtime to stall.

**Comment by appflowy** [comment]
> This is something that probably should be measured, at least in performance tests. Do we know how much size does Brotli saves us (after all it's optimised for HTML data not regular Yrs format), how long does it actually take and how other algorithms (ie. zstd) perform in comparison?

Yes. we can. But old fontend clients are still using Brotli to compress data.  when import a csv file that contains lots of rows.Each of them is a collab. Maybe I can add some metric that determine using rayon or not.

**Pull request merged**